File: apps/api/src/main/java/com/accordiq/dashboard/service/DashboardService.java
package com.accordiq.dashboard.service;

import com.accordiq.dashboard.dto.response.DashboardStatsResponse;
import com.accordiq.dashboard.dto.response.RecentDocumentResponse;

import java.util.List;

public interface DashboardService {

    DashboardStatsResponse getStatistics();

    List<RecentDocumentResponse> getRecentDocuments();

}

File: apps/web/services/dashboard.service.ts
import axios from "axios";

import type {
  DashboardStats,
  RecentDocument,
} from "@/types/dashboard";

const API =
  process.env.NEXT_PUBLIC_API_BASE_URL ??
  "http://localhost:8080/api/v1";

class DashboardService {

  async getStatistics(): Promise<DashboardStats> {

    const response = await axios.get(
      `${API}/dashboard/stats`
    );

    return response.data.data;

  }

  async getRecentDocuments(): Promise<
    RecentDocument[]
  > {

    const response = await axios.get(
      `${API}/dashboard/recent`
    );

    return response.data.data;

  }

}

export default new DashboardService();